modules/apps/object/object-api/src/main/java/com/liferay/object/field/util/ObjectFieldUtil.java

```diff
@@ -280,12 +283,38 @@ private static void _validateNewValue(
 			Object existingValue, String objectFieldName, Object value)
 		throws PortalException {
 
-		if (!((Validator.isNull(existingValue) && Validator.isNull(value)) ||
-			  Objects.equals(existingValue, value))) {
+		if (Validator.isNull(existingValue) && Validator.isNull(value)) {
+			return;
+		}
+
+		if ((value instanceof Boolean) &&
+			Objects.equals(
+				GetterUtil.getBoolean(value),
+				GetterUtil.getBoolean(existingValue))) {
 
-			throw new ObjectFieldReadOnlyException(
-				"Object field " + objectFieldName + " is read only");
+			return;
 		}
+
+		try {
+			BigDecimal bigDecimal1 = new BigDecimal(existingValue.toString());
```

Comment:
Can we test if `value` is an instance of BigDecimal here? I'm afraid of always hit the catch condition for all other types of fields.